pycodestyle.py

```diff
@@ -1662,9 +1663,8 @@ def check_logical(self):
             self.init_checker_state(name, argument_names)
             for offset, text in self.run_check(check, argument_names) or ():
                 if not isinstance(offset, tuple):
-                    for token_offset, pos in mapping:
-                        if offset <= token_offset:
-                            break
+                    token_offset, pos = mapping[bisect.bisect_left(
```

Comment:
I'd prefer it if you added a comment here explaining the benefit of `bisect_left` so this doesn't get replaced later on by a well-meaning person.